src/promptflow/promptflow/batch/_batch_engine.py

```
@@ -158,67 +151,112 @@ def run(
+        run_id: str = None,
+        output_dir: Path = None,
+        raise_on_line_failure: bool = False,
+    ) -> BatchResult:
+        try:
+            await self._executor_proxy.ensure_executor_health()
+            # apply default value in early stage, so we can use it both in line and aggregation nodes execution.
+            batch_inputs = [
+                apply_default_value_for_input(self._flow.inputs, each_line_input) for each_line_input in batch_inputs
+            ]
+            run_id = run_id or str(uuid.uuid4())
 
-    async def _exec_batch_internal(
+            # execute lines
+            if isinstance(self._executor_proxy, PythonExecutorProxy):
+                line_results.extend(self._executor_proxy._exec_batch(batch_inputs, output_dir, run_id))
+            else:
+                await self._exec_batch(line_results, batch_inputs, run_id)
+            handle_line_failures([r.run_info for r in line_results], raise_on_line_failure)
+
+            # execute aggregation nodes
+            aggr_exec_result = await self._exec_aggregation(batch_inputs, line_results, run_id)
+            # use the execution result to update aggr_result to make sure we can get the aggr_result in _exec_in_task
+            self._update_aggr_result(aggr_result, aggr_exec_result)
+
+            # persist outputs to output dir
+            outputs = [
+                {LINE_NUMBER_KEY: r.run_info.index, **r.output}
+                for r in line_results
+                if r.run_info.status == Status.Completed
+            ]
+            self._persist_outputs(outputs, output_dir)
+            # summary some infos from line results and aggr results to batch result
+            return BatchResult.create(self._start_time, datetime.utcnow(), line_results, aggr_result)
+        except asyncio.CancelledError:
+            # If the batch run is canceled, log it and ignore the exception.
+            bulk_logger.warning("The batch run is canceled.")
```

I think we should reraise this exception to avoid unexpected behavior.

---

Put try-catch to _exec_in_task

---

Updated.